testing/test_collection.py

```diff
@@ -1108,7 +1109,9 @@ def test_collect_pyargs_with_testpaths(testdir, monkeypatch):
     """
         )
     )
-    monkeypatch.setenv("PYTHONPATH", str(testdir.tmpdir))
+    pp = os.getenv("PYTHONPATH")
+    pp = ":".join((str(testdir.tmpdir), pp)) if pp else str(testdir.tmpdir)
```

Comment:
On Windows the path separator is actually `;`. Also, `setenv` has a `prepend` option which we can use to simplify this code to only this:

```python
monkeypatch.setenv("PYTHONPATH", str(testdir.tmpdir), prepend=os.pathsep)
```

👍

Reply:
Actually I think we can get away with using `monkeypatch.syspath_prepend` instead; I took the liberty of editing the file directly, I hope @hroncok doesn't mind. 😁

Reply:
I don't mind, I just want this fixed. This is much cleaner now, thanks.

Reply:
`monkeypatch.syspath_prepend()` only changes `sys.path`, not `PYTHONPATH` and this tests uses subprocess to call Python. Your fix doesn't seem to work.

Reply:
setenv with prepend should, will amend

Reply:
Back to:

```python
monkeypatch.setenv("PYTHONPATH", str(testdir.tmpdir), prepend=os.pathsep)
```